components/bl-worldClock-component/src/views/analog-clock.js

```diff
@@ -0,0 +1,107 @@
+import AnalogClockClassic from '../lib/analog-clock-classic';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const AnalogClockTypes = {
+  GOLD   : 'gold',
+  MODERN : 'modern',
+  VINTAGE: 'vintage',
+};
+const AnalogClockViews = {
+  [AnalogClockTypes.GOLD]   : AnalogClockGold,
+  [AnalogClockTypes.MODERN] : AnalogClockModern,
+  [AnalogClockTypes.VINTAGE]: AnalogClockVintage,
+};
+
+export function AnalogClock({ time, label, type, displaySeconds }) {
+  const { hour, minute, second } = time;
+
+  const Clock = AnalogClockViews[type];
+
+  return (
+    <div className={ cn('analog', type) }>
+      { Clock ? (
+        <Clock handStyle={ useHandStyles(hour, minute, second, displaySeconds) }/>
+      ) : (
```

Comment:
Component could exists without type here? How does it happens?

Reply:
The type for the component is defined above before the return